code/perception/src/coordinate_transformation.py

```diff
@@ -43,19 +26,39 @@ class CoordinateTransformer:
     h_ref: float
     ref_set = False
 
-    def __init__(self, gps_ref: GeoRef):
-        self.la_ref = gps_ref.value[0]
-        self.ln_ref = gps_ref.value[1]
-        self.h_ref = gps_ref.value[2]
+    def __init__(self):
+        pass
 
     def gnss_to_xyz(self, lat, lon, h):
-        return geodetic_to_enu(lat, lon, h,
-                               self.la_ref, self.ln_ref, self.h_ref)
+        return geodetic_to_enu(lat, lon, h)
+
+
+def geodetic_to_enu(lat, lon, alt):
+    """
+    Method from pylot project to calculate coordinates
+    https://github.com/erdos-project/pylot/blob/master/pylot/utils.py#L470
+
+    Args:
+        lat (float): latitude
+        lon (float): longitude
+        alt (float: altitude
+
+    Returns:
+        x, y, z: coordinates
+    """
+
+    scale = math.cos(CoordinateTransformer.la_ref * math.pi / 180.0)
+    basex = scale * math.pi * a / 180.0 * CoordinateTransformer.ln_ref
+    basey = scale * a * math.log(
+        math.tan((90.0 + CoordinateTransformer.la_ref) * math.pi / 360.0))
 
+    x = scale * math.pi * a / 180.0 * lon - basex
+    y = scale * a * math.log(
+        math.tan((90.0 + lat) * math.pi / 360.0)) - basey
 
-def geodetic_to_enu(lat, lon, h, lat_ref, lon_ref, h_ref):
-    x, y, z = geodetic_to_ecef(lat, lon, h)
-    return ecef_to_enu(x, y, z, lat_ref, lon_ref, h_ref)
+    # Is not necessary in new version
+    # y *= -1
```

Comment:
# alt + 331.00000 is needed, because the altitude of the map itself has to be accounted for
# TODO: maybe can be removed